spec/controllers/restaurants_controller_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe RestaurantsController, type: :controller do
+
+  describe 'Create new restaurant' do
+    it 'create a restaurant with valid parameters' do
+      post :create, params: FactoryGirl.attributes_for(:restaurant)
+      expect(response).to have_http_status(:created)
+    end
+
+    it 'fail when create a restaurant with missing address' do
+      post :create, params: FactoryGirl.attributes_for(:invalid_restaurant)
+      expect(response).to have_http_status(:unprocessable_entity)
+      end
+
+    end
+
+  describe "GET #index" do
+    it "get array of restaurants" do
+      get :index
+      expect(response).to have_http_status(:ok)
+    end
+
+    it "renders the :index view" do
+      get :index, params:  FactoryGirl.attributes_for(:restaurant)
+      expect(response).to have_http_status(:ok)
+    end
+  end
+
+  describe "GET #show" do
+    it "get valid rating of a restaurant" do
+      restaurant = FactoryGirl.attributes_for(:restaurant)
+      get :show,   params: { id: restaurant[:id] }
+      expect(response).to have_http_status(:ok)
+      expect(JSON.parse(response.body)["rating"]).to be_between(1, 5)
```

Comment:
testing for specific business rules (e.g. rating is 1-5) is superfluous in my view. those should be tested in the model (like you did below). assume that tomorrow ratings change to 1-10, preferably tests should only break at a single place (the model IMO).

Reply:
i think you can use the response helper `json` you've defined below